src/example-relay/src/mutations/RangeAdd/LocationsForm.js

```diff
@@ -0,0 +1,111 @@
+// @flow strict-local
+
+/* eslint-disable no-alert */
+
+import * as React from 'react';
+import { InputField, Stack, Button, Select } from '@kiwicom/orbit-components';
+import { useRelayEnvironment } from '@adeira/relay';
+
+import addLocation from './addLocation';
+import type { addLocationMutationResponse } from './__generated__/addLocationMutation.graphql';
+
+type Props = {||};
+
+type State = {|
+  +name: string,
+  +locationId: string,
+  +type: string,
+|};
+type Action =
+  | SyntheticInputEvent<HTMLInputElement>
+  | SyntheticInputEvent<HTMLSelectElement>
+  | 'reset';
+const initialState = { name: '', locationId: '', type: '' };
+function reducer(state: State, action: Action) {
+  if (action === 'reset') {
+    return initialState;
+  }
+  return {
+    ...state,
+    [action.target.name]: action.target.value,
+  };
+}
+
+const getLocation = (state: State) => {
+  // We have to map the type from string to enum
+  let type: 'AIRPORT' | 'CITY' | 'COUNTRY';
+  switch (state.type) {
+    case 'AIRPORT':
+    case 'CITY':
+    case 'COUNTRY':
+      type = state.type;
+      break;
+    default:
+      throw new Error('You selected a value not in the select field');
+  }
+  return {
+    ...state,
+    type,
+  };
+};
+export default (function LocationsForm() {
+  const [state, dispatch] = React.useReducer(reducer, initialState);
+  const [loading, setLoading] = React.useState(false);
+  const env = useRelayEnvironment();
+  const onSubmit = (e: SyntheticEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    if (state.type === '') {
+      alert('You must select a type');
+      return;
+    }
+    setLoading(true);
```

Comment:
You should use our `useMutation` hook instead. It returns `isPending` boolean so you don't have to deal with it manually like this.